File: ruby/quickstart.rb
# QuantumProxies — Ruby quickstart (net/http).
# Rotating residential proxies: new IP per request, sticky sessions, geo-targeting.
# Get credentials at https://quantumproxies.io
require "net/http"
require "json"
require "uri"

USER = "YOUR_USERNAME" # from your dashboard
PASS = "YOUR_PASSWORD"
HOST = "residentialboson.quantumproxies.io"

def through_proxy(port:, user:)
  uri = URI("https://ipinfo.io/json")
  http = Net::HTTP.new(uri.host, uri.port, HOST, port, user, PASS)
  http.use_ssl = true
  http.open_timeout = http.read_timeout = 30
  JSON.parse(http.get(uri.path).body)
end

# Rotating: new IP on every request (port 9000)
puts "rotating: #{through_proxy(port: 9000, user: USER)["ip"]}"

# Country targeting: append -country-XX to the username
geo = through_proxy(port: 9000, user: "#{USER}-country-fr")
puts "fr exit:  #{geo["ip"]} #{geo["country"]}"

# Sticky session: same IP for `lifetime` minutes (port 10000)
sticky = "#{USER}-country-us-session-ab12-lifetime-10"
puts "sticky #1: #{through_proxy(port: 10_000, user: sticky)["ip"]}"
puts "sticky #2: #{through_proxy(port: 10_000, user: sticky)["ip"]}" # same IP

File: java/QuickStart.java
// QuantumProxies — Java quickstart (java.net.http, JDK 11+).
// Rotating residential proxies: new IP per request, sticky sessions, geo-targeting.
// Get credentials at https://quantumproxies.io
//
//   javac QuickStart.java && java -Djdk.http.auth.tunneling.disabledSchemes= QuickStart

import java.net.Authenticator;
import java.net.InetSocketAddress;
import java.net.PasswordAuthentication;
import java.net.ProxySelector;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class QuickStart {
    static final String USER = "YOUR_USERNAME"; // from your dashboard
    static final String PASS = "YOUR_PASSWORD";
    static final String HOST = "residentialboson.quantumproxies.io";

    static String through(String user, int port) throws Exception {
        HttpClient client = HttpClient.newBuilder()
            .proxy(ProxySelector.of(new InetSocketAddress(HOST, port)))
            .authenticator(new Authenticator() {
                @Override protected PasswordAuthentication getPasswordAuthentication() {
                    return new PasswordAuthentication(user, PASS.toCharArray());
                }
            })
            .build();
        HttpRequest req = HttpRequest.newBuilder(URI.create("https://ipinfo.io/json")).build();
        return client.send(req, HttpResponse.BodyHandlers.ofString()).body();
    }

    public static void main(String[] args) throws Exception {
        // Rotating: new IP on every request (port 9000)
        System.out.println("rotating: " + through(USER, 9000));

        // Country targeting: append -country-XX to the username
        System.out.println("it exit: " + through(USER + "-country-it", 9000));

        // Sticky session: same IP for `lifetime` minutes (port 10000)
        String sticky = USER + "-country-us-session-ab12-lifetime-10";
        System.out.println("sticky #1: " + through(sticky, 10000));
        System.out.println("sticky #2: " + through(sticky, 10000)); // same IP
    }
}
